lightly/transforms/fda_transform.py

```diff
@@ -0,0 +1,246 @@
+from typing import Optional, Tuple, Union
+
+from PIL.Image import Image
+from torch import Tensor
+
+from lightly.transforms.amplitude_rescale_transform import AmplitudeRescaleTransform
+from lightly.transforms.gaussian_blur import GaussianBlur
+from lightly.transforms.gaussian_mixture_masks_transform import GaussianMixtureMask
+from lightly.transforms.irfft2d_transform import IRFFT2DTransform
+from lightly.transforms.multi_view_transform import MultiViewTransform
+from lightly.transforms.phase_shift_transform import PhaseShiftTransform
+from lightly.transforms.random_frequency_mask_transform import (
+    RandomFrequencyMaskTransform,
+)
+from lightly.transforms.rfft2d_transform import RFFT2DTransform
+from lightly.transforms.solarize import RandomSolarization
+from lightly.transforms.torchvision_v2_compatibility import torchvision_transforms as T
+
+
+class FDAView1Transform:
+    def __init__(
+        self,
+        # Random resized crop
+        input_size: int = 224,
+        min_scale: float = 0.08,
+        # Color jitter
+        cj_prob: float = 0.8,
+        cj_contrast: float = 0.4,
+        cj_bright: float = 0.4,
+        cj_sat: float = 0.2,
+        cj_hue: float = 0.1,
+        cj_strength: float = 1.0,
+        # Grayscale
+        random_gray_scale: float = 0.2,
+        # Gaussian blur
+        gaussian_blur: float = 1.0,
+        sigmas: Tuple[float, float] = (0.1, 2),
+        kernel_size: Optional[float] = 23,
+        # Amplitude rescale
+        ampl_rescale_range: Tuple[float, float] = (0.8, 1.75),
+        amp_rescale_prob: float = 0.2,
+        # Phase shift
+        phase_shift_range: Tuple[float, float] = (0.4, 0.7),
+        phase_shift_prob: float = 0.2,
+        # Random frequency mask
+        rand_freq_mask_range: Tuple[float, float] = (0.01, 0.1),
+        rand_freq_mask_prob: float = 0.5,
+        # Gaussian mixture mask
+        num_gaussians: int = 20,
+        std_range: Tuple[float, float] = (10, 15),
+        gm_prob: float = 0.2,
+        # Other
+        hf_prob: float = 0.5,
+        solarization_prob: float = 0.0,
```

Comment:
Let's also add the others even though they are optional:
```
        vf_prob: float = 0.0,
        hf_prob: float = 0.5,
        rr_prob: float = 0.0,
        rr_degrees: Optional[Union[float, Tuple[float, float]]] = None,
```

This just makes all lightly transforms a bit more consistent and adds flexibility for other use-cases than in the paper.